xblock/core.py

```diff
@@ -1,83 +1,121 @@
 from xblock.validation import Validation
 
-# exposing XML_NAMESPACES as a member of core, in order to avoid importing mixins where
-# XML_NAMESPACES are needed (e.g. runtime.py).
-XML_NAMESPACES = xblock.mixins.XML_NAMESPACES
+# OrderedDict is used so that namespace attributes are put in predictable order
+# This allows for simple string equality assertions in tests and have no other effects
+XML_NAMESPACES = OrderedDict([
+    ("option", "http://code.edx.org/xblock/option"),
+    ("block", "http://code.edx.org/xblock/block"),
+])
 
 # __all__ controls what classes end up in the docs.
-__all__ = ['XBlock']
+__all__ = ['XBlock', 'XBlockAside']
+
 UNSET = object()
 
 
-class XBlockMixin(ScopedStorageMixin):
+class _AutoNamedFieldsMetaclass(type):
     """
-    Base class for XBlock Mixin classes.
+    Builds classes such that their Field attributes know their own names.
 
-    XBlockMixin classes can add new fields and new properties to all XBlocks
-    created by a particular runtime.
+    This allows XBlock API users to define fields without name redundancy, e.g. like this:
 
-    """
+        class MyBlock(XBlock):
+            my_field = Field(...)
 
+    rather than this:
 
-class SharedBlockBase(Plugin):
+        class MyBlock(XBlock):
+            my_field = Field(name="my_field", ...)
     """
-    Behaviors and attrs which all XBlock like things should share
+    def __new__(mcs, name, bases, attrs):
+        """
+        Ensure __name__ is set on all Field attributes, both on the new class and on its bases.
+        """
+        def needs_name(obj):
+            """
+            Is this object a Field that hasn't had its name assigned yet?
+            """
+            return isinstance(obj, Field) and not obj.__name__
+
+        # Iterate over the attrs before they're bound to the class
+        # so that we don't accidentally trigger any __get__ methods
+        for attr_name, attr in attrs.items():
+            if needs_name(attr):
+                attr.__name__ = attr_name
+
+        # Iterate over all of the base classes, so that we can add
+        # names to any mixins that don't include this metaclass, but that
+        # do include Fields
+        for base in bases:
+            for attr_name, attr in inspect.getmembers(base, predicate=needs_name):
+                attr.__name__ = attr_name
+
+        return super().__new__(mcs, name, bases, attrs)
+
+
+class Blocklike(metaclass=_AutoNamedFieldsMetaclass):
     """
+    Shared base for XBlocks and XBlockAsides, providing these common capabilities:
+
+    - services
+    - fields
+    - OLX
+    - handlers
+    - resources
 
+    (see XBlock and XBlockAside classes for details)
+    """
     resources_dir = ''
     public_dir = 'public'
     i18n_js_namespace = None
 
     @classmethod
     def get_resources_dir(cls):
         """
-        Gets the resource directory for this XBlock.
+        Gets the resource directory for this XBlock-like class.
```

Comment:
I've tried to generalize these docstrings because they apply to both XBlocks and XBlockAsides. Hopefully "XBlock-like class" isn't too clunky for folks reading the API docs.